lib/supavisor/logger/filters.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Supavisor.Logger.Filters do
+  @moduledoc """
+  Useful logger filters.
+  """
+
+  @doc """
+  Log events that are fired by `Supavisor.ClientHandler` only when the module
+  state is equal to `state`.
+  """
+  def filter_client_handler(log_event, state) do
+    %{meta: meta} = log_event
+
+    with %{mfa: {Supavisor.ClientHandler, _, _}, state: curr_state} <- meta,
+         true <- curr_state == state do
+      log_event
+    else
+      _ -> :ignore
+    end
+  end
```

Comment:
maybe using clauses could be a bit faster

```suggestion
  def filter_client_handler(
        %{meta: %{mfa: {Supavisor.ClientHandler, _, _}, state: ^state}} = log_event,
        state
      ) do
    log_event
  end

  def filter_client_handler(_, _), do: :ignore
```

Reply:
In the end it will be compiled to similar code, so I do not think that this is really needed.

Reply:
Although I read everywhere that PM via clauses is faster, I need to benchmark it sometime